core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -323,13 +323,11 @@ public void update(Supplier<JournalContext> journalContext, AlluxioURI alluxioUr
   public String getMountPoint(AlluxioURI uri) throws InvalidPathException {
     String path = uri.getPath();
     String lastMount = ROOT;
+    List<String> possibleMounts = PathUtils.getPossibleMountPoints(path);
     try (LockResource r = new LockResource(mReadLock)) {
-      for (Map.Entry<String, MountInfo> entry : mState.getMountTable().entrySet()) {
-        String mount = entry.getKey();
-        // we choose a new candidate path if the previous candidate path is a prefix
-        // of the current alluxioPath and the alluxioPath is a prefix of the path
-        if (!mount.equals(ROOT) && PathUtils.hasPrefix(path, mount)
-            && lastMount.length() < mount.length()) {
+      Map<String, MountInfo> mountTable = mState.getMountTable();
+      for (String mount: possibleMounts) {
+        if (mountTable.containsKey(mount)) {
```

Comment:
I think the original algorithm can handle the recursive check, it will take the longest mount. Looks like your change would miss that part.

Reply:
This PR also meets this part. The results returned by `getPossibleMountPoints` are from shortest to longest. I will add a comment to make this clear.

Reply:
Added.